React/Base/RCTConvert.m

```diff
@@ -567,6 +571,25 @@ static BOOL RCTFontIsCondensed(UIFont *font)
   return (symbolicTraits & UIFontDescriptorTraitCondensed) != 0;
 }
 
+NSDictionary *RCTFontVariantMapping = NULL;
+
++ (NSDictionary *)getRCTFontVariantMapping
+{
+  if (!RCTFontVariantMapping) {
+    RCTFontVariantMapping = @{
+      @"tabular-nums": @{
```

Comment:
The ones implemented are mutually exclusive, but you can also have small caps, ligatures etc. But you do have a point—maybe there should be a bit more validation.